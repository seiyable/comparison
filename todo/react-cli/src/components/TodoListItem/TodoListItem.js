import React from 'react';
import styles from './TodoListItem.module.css';

export default class TodoListItem extends React.Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
      <li className={styles.todoListItem}>
        <span>{ this.props.todo }</span>
        <button
          type="button"
          onClick={() => this.props.removeTodo(this.props.index)}>
          DONE!
        </button>
      </li>
    );
  }
}
