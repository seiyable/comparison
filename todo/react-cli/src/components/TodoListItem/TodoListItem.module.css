.todoListItem {
  display: flex;
  align-items: center;
  margin-bottom: 10px;
}

.todoListItem button {
  margin-left: 15px;
}
