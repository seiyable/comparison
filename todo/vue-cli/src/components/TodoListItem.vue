<template>
  <li class="todo-list-item">
    <span>{{ todo }}</span>
    <button
      type="button"
      class="done"
      @click="onDoneButtonClick">
      DONE!
    </button>
  </li>
</template>

<script>
export default {
  props: {
    todo: {
      type: String,
      default: '',
    },
    index: {
      type: Number,
      required: true,
    },
  },
  methods: {
    // called when DONE! button is clicked
    onDoneButtonClick() {
      this.$emit('done', this.index);
    },
  },
};
</script>

<style lang="stylus" scoped>
@import '~@/stylus/variables'

li.todo-list-item
  display: flex
  align-items: center
  margin-bottom: ($gap / 2)

  button.done
    margin-left: ($gap * 3 / 4)
</style>
