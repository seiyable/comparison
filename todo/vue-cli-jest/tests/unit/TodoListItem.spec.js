import {shallowMount} from '@vue/test-utils';
import TodoListItem from '@/components/TodoListItem.vue';

const factory = (values = {}) => {
  return shallowMount(TodoListItem, {
    propsData: {...values},
  });
};

describe('TodoListItem', () => {
  it('renders todo text received from props', () => {
    const sampleTodo = 'yoga';
    const wrapper = factory({
      todo: sampleTodo,
      index: 0,
    });

    expect(wrapper.text()).toContain(sampleTodo);
  });

  it('emits "done" event on button click', () => {
    const sampleTodo = 'yoga';
    const wrapper = factory({
      todo: sampleTodo,
      index: 0,
    });

    const button = wrapper.find('.done');
    button.trigger('click');

    expect(wrapper.emitted()['done']).toBeTruthy();
  });
});
