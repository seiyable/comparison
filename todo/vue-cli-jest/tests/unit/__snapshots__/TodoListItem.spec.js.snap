// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListItem renders correctly 1`] = `
<li
  class="todo-list-item"
>
  <span>
    yoga
  </span>
   
  <button
    class="done"
    type="button"
  >
    
    DONE!
  
  </button>
</li>
`;
